docs/adding-a-language.md

```
@@ -0,0 +1,843 @@
+Run it — it should fail to compile (`SwiftParser` doesn't exist):
+
+```bash
+cd dependi-lsp
+cargo test parsers::swift
+```
+
+Expected: compilation error mentioning `cannot find type SwiftParser`.
+
+### 4.3 Implement
+
+Replace the rest of `dependi-lsp/src/parsers/swift.rs` body with the implementation. The doctest [Example 3](#example-3--implementing-the-parser-trait) on this page contains a complete implementation you can copy. The full file:
```

markdownlint-cli2 (0.22.1)

[warning] Link fragments should be valid

(MD051, link-fragments)